drivers/video/video_shell.c

```diff
@@ -0,0 +1,654 @@
+	size_t rows = cols * height / width;
+	size_t debt = 0;
+
+	if (cols > sizeof(line[0])) {
+		shell_error(sh, "parameter <columns> (%u) larger than line buffer %u",
+			    cols, sizeof(line[0]));
+		return -EINVAL;
+	}
+
+	if (pixfmt != VIDEO_PIX_FMT_RGB565) {
+		shell_error(sh, "only <fourcc> 'RGBP' is supported for now");
+		return -ENOTSUP;
+	}
+
+	for (size_t h = 0, r = 0, i = 0; i < size; i = pitch * h) {
+		/* Convert this entire row */
+		video_line_rgb565le_to_vt100((uint16_t *)&vbuf->buffer[i], width, line[r % 2], cols);
+
+		/* Skip the bytes instead of scaling: poor quality but faster */
+		video_skip_to_next(&r, rows, &h, height, &debt);
+
+		/* Once every two rows, display the result */
+		if (r % 2 == 0) {
+			for (size_t c = 0; c < cols; c++) {
+				shell_fprintf_normal(sh, "\e[48;5;%u;38;5;%um▄\e[m", line[0][c], line[1][c]);
+			}
+			shell_fprintf_normal(sh, ":\n");
+		}
+	}
+
+	return 0;
+}
+
+static int cmd_video_view(const struct shell *sh, size_t argc, char **argv)
+{
+	size_t pixfmt = 0;
+	size_t width = 0;
+	size_t offset = 0;
+	size_t cols = 100;
+	size_t size;
+	char *endptr;
+
+	if (vbuf == NULL) {
+		shell_error(sh, "buffer not allocated yet, use the 'alloc' command first");
+		return -ENOBUFS;
+	}
+
+	size = vbuf->bytesused;
+
+	pixfmt = video_shell_parse_fourcc(sh, argv[1]);
+	if (pixfmt == 0 || video_pix_fmt_bpp(pixfmt) == 0) {
+		shell_error(sh, "unsupported <fourcc> %s", argv[1]);
+		return -EINVAL;
+	}
+
+	width = strtoul(argv[2], &endptr, 10);
+	if (*endptr != '\0' || width == 0) {
+		shell_error(sh, "invalid <width> parameter %s", argv[2]);
+		return -EINVAL;
+	}
+
+	for (size_t argidx = 3; argidx < argc; argidx++) {
+		if (argidx + 1 >= argc) {
+			shell_error(sh, "missing argument after command flag");
+			return -EINVAL;
+		} else if (strcmp(argv[argidx], "-s") == 0) {
+			argidx++;
+			size = strtoul(argv[argidx], &endptr, 10);
+			if (*endptr != '\0' || size == 0) {
+				shell_error(sh, "invalid <size> parameter %s", argv[argidx]);
+				return -EINVAL;
+			}
+		} else if (strcmp(argv[argidx], "-o") == 0) {
+			argidx++;
+			offset = strtoul(argv[argidx], &endptr, 10);
+			if (*endptr != '\0') {
+				shell_error(sh, "invalid <offset> parameter", argv[argidx]);
+				return -EINVAL;
+			}
+		} else if (strcmp(argv[argidx], "-c") == 0) {
+			argidx++;
+			cols = strtoul(argv[argidx], &endptr, 10);
+			if (*endptr != '\0') {
+				shell_error(sh, "invalid <width> parameter %s", argv[argidx]);
+				return -EINVAL;
+			}
+		} else {
+			shell_error(sh, "unknown argument %s", argv[argidx]);
+			return -EINVAL;
+		}
+	}
+
+	if (offset > vbuf->bytesused) {
+		shell_error(sh, "offset (%u) larger than data (%u)", offset, vbuf->bytesused);
+		return -EINVAL;
+	}
+
+	return video_shell_view(sh, &vbuf->buffer[offset], size - offset, pixfmt, width, cols);
+}
+
+static void dcmd_video_device_name(size_t idx, struct shell_static_entry *entry)
+{
+	const struct device *dev = shell_device_lookup(idx, NULL);
```

Comment:
you can make this smarter and only display video devices :)
https://github.com/zephyrproject-rtos/zephyr/blob/84532b24e54e9ba205a429f004070e59e6c0105d/drivers/led/led_shell.c#L332-L341

Reply:
Nice!
```
uart:~$ video start
  VIDEO_SW_GENERATOR  video_emul_imager   video_emul_rx
uart:~$ video start
```